I made a mistake by not adding `--ignore-environment` with `nix develop`. Build failure persist with current pr. But things are getting even weirder. Errors are different between doing manual build with `genericBuild` and `nix build`. And no error with manual `make`.

If I run `nix build .#moarvm`, I get:

```
......

ld: warning: ignoring file 3rdparty/dyncall/dyncallback/libdyncallback_s.a, building for macOS-x86_64 but attempting to li>
ld: warning: ignoring file 3rdparty/dyncall/dynload/libdynload_s.a, building for macOS-x86_64 but attempting to link with >
ld: warning: ignoring file 3rdparty/dyncall/dyncall/libdyncall_s.a, building for macOS-x86_64 but attempting to link with >
Undefined symbols for architecture x86_64:

......
```

Inside the shell spawned by `nix develop .#moarvm --ignore-environment`, if I build moarvm with `buildPhase` or `genericBuild`, the following errors will occur:

```
......

linking libmoar.dylib
ld: warning: ignoring file 3rdparty/sha1/libsha1.a, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x2F 0x20 0x20 0x20 0x20 0x20 0x20 0x20 )
ld: warning: ignoring file 3rdparty/cmp/libcmp.a, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x2F 0x20 0x20 0x20 0x20 0x20 0x20 0x20 )
ld: warning: ignoring file 3rdparty/libuv/libuv.a, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x2F 0x20 0x20 0x20 0x20 0x20 0x20 0x20 )
ld: warning: ignoring file 3rdparty/libtommath/libtommath.a, building for macOS-x86_64 but attempting to link with file built for macOS-x86_64

......
```

But if I manually build it with `make` after unpack, patch, configure phase, it will not give any error.

I have no clue how this would happen. Did I do something wrong or is nix on x86_64-darwin kind of broken?

--- a/pkgs/development/interpreters/rakudo/moarvm.nix
+++ b/pkgs/development/interpreters/rakudo/moarvm.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, cctools-port
 , perl
 , CoreServices
 , ApplicationServices
@@ -28,6 +29,10 @@ stdenv.mkDerivation rec {
     substituteInPlace 3rdparty/dyncall/configure \
       --replace '`sw_vers -productVersion`' '"$MACOSX_DEPLOYMENT_TARGET"'
   '';
+
+  nativeBuildInputs = lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
+    cctools-port
+  ];
 
   buildInputs = [ perl ] ++ lib.optionals stdenv.isDarwin [ CoreServices ApplicationServices ];
   doCheck = false; # MoarVM does not come with its own test suite

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -17970,6 +17970,7 @@ with pkgs;
 
   rakudo = callPackage ../development/interpreters/rakudo { };
   moarvm = callPackage ../development/interpreters/rakudo/moarvm.nix {
+    inherit (darwin) cctools-port;
     inherit (darwin.apple_sdk.frameworks) CoreServices ApplicationServices;
   };
   nqp = callPackage  ../development/interpreters/rakudo/nqp.nix { };

